### PR TITLE
Data center reachability in cluster state

### LINKED ISSUE #23347: Add data center reachability to ClusterReadView
Should be done in a backwards compatible way after https://github.com/akka/akka/pull/23293

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #23347

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala [hunk 1]
```diff
@@ -125,6 +161,47 @@ object ClusterEvent {
...
+    // Product5
+    override def productPrefix = "CurrentClusterState"
+    def _1: SortedSet[Member] = members
+    def _2: Set[Member] = unreachable
+    def _3: Set[Address] = seenBy
+    def _4: Option[Address] = leader
+    def _5: Map[String, Option[Address]] = roleLeaderMap
+    def canEqual(that: Any): Boolean = that.isInstanceOf[CurrentClusterState]
+
+    override def toString = s"CurrentClusterState($members, $unreachable, $seenBy, $leader, $roleLeaderMap)"
```
Comment (ktoso) [inline akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala]: When doing this horrible case class -> class I usually do $_1 etc to make sure order remains as it was... Perhaps that's a terrible idea by me though?

File: akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala [hunk 2]
```diff
@@ -125,6 +161,47 @@ object ClusterEvent {
...
+
+    // Product5
+    override def productPrefix = "CurrentClusterState"
+    def _1: SortedSet[Member] = members
+    def _2: Set[Member] = unreachable
+    def _3: Set[Address] = seenBy
+    def _4: Option[Address] = leader
+    def _5: Map[String, Option[Address]] = roleLeaderMap
+    def canEqual(that: Any): Boolean = that.isInstanceOf[CurrentClusterState]
+
```
Comment (ktoso) [inline akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala]: yup, that's it

File: akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala (hunk 1 above)
Comment (johanandren) [inline akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala]: I started out like that but then I thought it looked so ugly I didn't want to keep that.

Comment (ktoso) [inline akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala]: Yeah, I think you're right on that

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**